feat(api): update `init` method of discord integration

Arnesh-pal opened the pull request:
### **User description**
## Description

Implemented Discord initialization confirmation message

Added logic in discord.integration.ts to send a confirmation message ("Drum rolls! Keyshade is now configured with this channel") to the Discord channel upon initialization.

Replaced the TODO placeholder with an async init() method that uses the existing makeTimedRequest utility for consistent HTTP handling.

Includes error handling and logging for failed requests.

Fixes #1081 

<br />

## Dependencies

Uses native fetch API (or node-fetch if running in Node.js).

Relies on existing PrismaService and makeTimedRequest from the codebase.

## Future Improvements

Integration Run Tracking: Consider logging initialization attempts in the database (like emitEvent does) for better debugging.

Customizable Message: Allow workspace admins to configure the initialization message via metadata.

Retry Mechanism: Add retries for transient Discord API failures.

## Screenshots of relevant screens

For my own channel 

I got 
<img width="1058" height="168" alt="image" src="https://github.com/user-attachments/assets/5abb2494-f6ad-4e00-bbe7-190226af0ba3" />

Using
curl -X POST -H "Content-Type: application/json" -d '{"content": "Drum rolls! Keyshade is now configured with this channel"}' [discord-webhook-url]

## Developer's checklist

- [✅] My PR follows the style guidelines of this project
- [✅] I have performed a self-check on my work

**If changes are made in the code:**

- [✅ ] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [✅] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [✅] I have added relevant screenshots in my PR
- [✅] There are no UI/UX issues


### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [✅] I have made the necessary updates to the documentation, or no documentation changes are required.


___

### **PR Type**
Enhancement


___

### **Description**
- Implemented Discord initialization confirmation message

- Added async init() method with error handling

- Replaced TODO placeholder with functional implementation

- Uses makeTimedRequest utility for consistent HTTP handling


___

### Diagram Walkthrough


```mermaid
flowchart LR
  A["Discord Integration"] --> B["init() method"]
  B --> C["Send POST request"]
  C --> D["Discord webhook"]
  D --> E["Confirmation message"]
  E --> F["Channel notification"]
```



<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>discord.integration.ts</strong><dd><code>Implement Discord initialization confirmation message</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/api/src/integration/plugins/discord.integration.ts

<ul><li>Replaced TODO placeholder with async init() method<br> <li> Added Discord webhook POST request with confirmation message<br> <li> Implemented error handling and logging for initialization<br> <li> Uses makeTimedRequest utility for consistent HTTP handling</ul>


</details>


  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/1083/files#diff-aed350d3ddd9f34357c7dd8e4b005b71d3eb6c4910d79816ed53c625b79ed5aa">+28/-3</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

rajdip-b commented on apps/api/src/integration/plugins/discord.integration.ts:
```diff
@@ -54,11 +54,36 @@ export class DiscordIntegration extends BaseIntegration {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        content: 'Drum rolls! Keyshade is now configured with this channel'
```
Would be better to replace Drum rolls! with actual drum rolls emote

rajdip-b commented on apps/api/src/integration/plugins/discord.integration.ts:
```diff
@@ -54,11 +54,36 @@ export class DiscordIntegration extends BaseIntegration {
-    return Promise.resolve()
+// Edited Code starts for init 
+  public async init(): Promise<void> {
+  const integration = this.getIntegration<DiscordIntegrationMetadata>()
```
This should be also creating an integration run. refer https://github.com/keyshade-xyz/keyshade/blob/fbdfb2740a6178eb8031b1f7d158fddc5a9c7b1a/apps/api/src/integration/plugins/discord.integration.ts#L92

rajdip-b commented on apps/api/src/integration/plugins/discord.integration.ts:
```diff
@@ -54,11 +54,36 @@ export class DiscordIntegration extends BaseIntegration {
+  this.logger.log('Successfully initialized Discord integration')
+}
+
+// Edited Code ends for init 
```
Remove this

Arnesh-pal commented:
Commit made just now for the same issue 
Added an emoji of drum roll
Added  integration run
And removed unnecessary print statements

rajdip-b commented on apps/api/src/integration/plugins/discord.integration.ts:
```diff
@@ -58,28 +58,34 @@ export class DiscordIntegration extends BaseIntegration {
   
-  this.logger.log('Initializing Discord integration')
+  const { id: integrationRunId } = await this.registerIntegrationRun({
+    eventId: 'init', // Special identifier for initialization
```
Event ID should come from the trigger event. Refer https://github.com/keyshade-xyz/keyshade/blob/9fba34814557a0fef4622c6ca3c820d3cd9d038f/apps/api/src/integration/plugins/vercel.integration.ts#L59-L62

rajdip-b commented on apps/api/src/integration/plugins/discord.integration.ts:
```diff
@@ -58,28 +58,34 @@ export class DiscordIntegration extends BaseIntegration {
       },
       body: JSON.stringify({
-        content: 'Drum rolls! Keyshade is now configured with this channel'
+        content: '🥁 Keyshade is now configured with this channel'
```
add !

Arnesh-pal replied:
Sorry didn't understand
what to add

Arnesh-pal replied:
Oh the exclamation mark
Understood

Arnesh-pal commented:
Added eventId parameter to match the pattern used in Vercel integration

Updated the message to include both the drum emoji and exclamation mark: '🥁 Keyshade is now configured with this channel!'

Arnesh-pal commented:
Can you try this change

rajdip-b commented:
Try running the e2e tests on your end. it's failing
